### ISSUE TITLE
Initial connection timeout?

### ISSUE THREAD

**Issue by MatthewMah** [TakahikoKawasaki/nv-websocket-client#10]
Is there currently the capability to set the initial connection timeout for the socket connection? I think this would use the socket method `connect(SocketAddress endpoint, int timeout) `. 

My code fragment looks like this, and for timeout_ms =6000, the timeout for the initial connection is highly variable, but can definitely be > 30 sec. The SO_TIMEOUT does not seem to have any effect on the initial connection timeout.  
```
websocket = factory.createSocket(connectionURI);
Socket s = websocket.getSocket();
s.setSoTimeout(timeout_ms);
websocket.addListener(this);
websocket.connect();
```

**Comment by TakahikoKawasaki** [comment]
Are you using a proxy? The current implementation performs negotiation with the specified proxy in `createSocket()`. If you are using a proxy and you need to set SO_TIMEOUT before the negotiation with the proxy, I'll improve the implementation.

On the other hand, if you are not using a proxy, show me the URL you are trying to access. I'll investigate the issue.

**Comment by MatthewMah** [comment]
I am not using a proxy. I was encountering the initial timeout situation with a server on a local network, 10.x.x.x. I disconnected the server from the network and tried to connect the client. Obviously the client will not be able to connect when the server is not connected to the network, but the timeout to recognize this was much greater than the 6 seconds I am trying to configure.

**Comment by TakahikoKawasaki** [comment]
I understood what the implementation has to do for your request just now. The implementation of `WebSocketFactory` needs to be changed. The current implementation calls `SocketFactory.createSocket(String host, int port)`, but the call has to be divided into the following two calls for _timeout_.

`SocketFactory.createSocket()`
`Socket.connect(new InetSocketAddress(host, port), timeout)`

And a variant of `WebSocket.connect` method which has a timeout argument needs to be added.

Give me some time for the change.

**Comment by TakahikoKawasaki** [comment]
Added some `createSocket` method variants which take a timeout value for socket connection.

- `WebSocketFactory.createSocket(String uri, int timeout)`
- `WebSocketFactory.createSocket(URL url, int timeout)`
- `WebSocketFactory.createSocket(URI uri, int timeout)`

The following is an excerpt from the new `README.md`.

<hr>
There are two ways to set a timeout value for socket connection. The first way is to call `setConnectionTimeout(int timeout)` method of `WebSocketFactory`.

```java
// Create a web socket factory and set 5000 milliseconds as a timeout
// value for socket connection.
WebSocketFactory factory = new WebSocketFactory().setConnectionTimeout(5000);

// Create a web socket. The timeout value set above is used.
WebSocket ws = factory.createSocket("ws://localhost/endpoint");
```

The other way is to give a timeout value to a `createSocket` method.

```java
// Create a web socket factory. The timeout value remains 0.
WebSocketFactory factory = new WebSocketFactory();

// Create a web socket with a socket connection timeout value.
WebSocket ws = factory.createSocket("ws://localhost/endpoint", 5000);
```

The timeout value is passed to `connect(SocketAddress, int)` method of `java.net.Socket`.

**Comment by TakahikoKawasaki** [comment]
I released a new version, 1.10, just now. The new version contains the changes described above.

**Comment by MatthewMah** [comment]
This is the fastest I have ever seen an open source software issue resolved. Thanks, and good work.
